lib/lockmaster.js

```diff
@@ -109,10 +110,17 @@ Lockmaster.prototype.start = function start() {
         };
 
         // Start running at intervals.
-        me._running[river] = setInterval(riverRunner, intervalSeconds);
+        if (river.config.hasOwnProperty('cronInterval')) {
+            _.each(river.config.cronInterval, function(interval) {
+                schedule.scheduleJob(interval, riverRunner);
+            });
+        } else {
+            me._running[river] = setInterval(riverRunner, intervalSeconds);
+
+            // Run once now at startup.
+            riverRunner();
```

Comment:
That is true, you are right, carry on. :)

---------
Matt Taylor
OS Community Flag-Bearer
Numenta

On Wed, Jul 29, 2015 at 9:55 AM, Jared Weiss <notifications@github.com>
wrote:

> In lib/lockmaster.js
> <https://github.com/nupic-community/river-view/pull/45#discussion_r35783207>
> :
>
> > @@ -109,10 +110,17 @@ Lockmaster.prototype.start = function start() {
> >          };
> >
> >          // Start running at intervals.
> > -        me._running[river] = setInterval(riverRunner, intervalSeconds);
> > +        if (river.config.hasOwnProperty('cronInterval')) {
> > +            _.each(river.config.cronInterval, function(interval) {
> > +                schedule.scheduleJob(interval, riverRunner);
> > +            });
> > +        } else {
> > +            me._running[river] = setInterval(riverRunner, intervalSeconds);
> > +
> > +            // Run once now at startup.
> > +            riverRunner();
>
> I wasn't sure if we wanted a river to run if the server was started
> outside of the cron interval or not in time with it (i.e. it's set to run
> every hour on the 15 minute mark and you start the server at 9:30). This
> way, only the regularly running rivers are run at startup since
> setInterval doesn't run immediately.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nupic-community/river-view/pull/45/files#r35783207>.
>